src/data/store/slices/linkSaturation.slice.ts

```diff
@@ -155,58 +150,112 @@ const linkSaturationSlice = createSlice({
-            if (linkState.type === LinkType.DRAM) {
-                linkState.maxBandwidth = DRAMBandwidthBytes / CLKHz;
-                recalculateLinkSaturation(linkState, graphState.totalOps);
-            }
+    Object.values(state.linksPerTemporalEpoch).forEach((epochState) => {
+        Object.entries(epochState.linksPerNodeMap).forEach(([nodeUid, { links, offchipLinkIds }]) => {
+            Object.values(links).forEach((linkState) => {
+                if (linkState.type === LinkType.DRAM) {
+                    linkState.maxBandwidth = DRAMBandwidthBytes / CLKHz;
+                    const { bpc, linkSaturation, nodeSaturation, offChipSaturation } = recalculateLinkSaturation({
+                        link: linkState,
+                        totalOpCycles: epochState.totalOps,
+                        links,
+                        offchipLinkIds,
+                    });
+
+                    linkState.bpc = bpc;
+                    linkState.saturation = linkSaturation;
+                    epochState.linksPerNodeMap[nodeUid].saturation = nodeSaturation;
+                    epochState.linksPerNodeMap[nodeUid].offchipSaturation = offChipSaturation;
+                }
+            });
         });
     });
 };
 
 const updatePCILinks = (state: NetworkCongestionState) => {
     const { PCIBandwidthGBs, CLKHz } = getInitialCLKValues(state);
 
-    Object.values(state.graphs).forEach((graphState) => {
-        Object.values(graphState.links).forEach((linkState: LinkState) => {
-            if (linkState.type === LinkType.PCIE) {
-                linkState.maxBandwidth = PCIBandwidthGBs / CLKHz;
-                recalculateLinkSaturation(linkState, graphState.totalOps);
-            }
+    Object.values(state.linksPerTemporalEpoch).forEach((epochState) => {
+        Object.entries(epochState.linksPerNodeMap).forEach(([nodeUid, { links, offchipLinkIds }]) => {
+            Object.values(links).forEach((linkState) => {
+                if (linkState.type === LinkType.PCIE) {
+                    linkState.maxBandwidth = PCIBandwidthGBs / CLKHz;
+                    const { bpc, linkSaturation, nodeSaturation, offChipSaturation } = recalculateLinkSaturation({
+                        link: linkState,
+                        totalOpCycles: epochState.totalOps,
+                        links,
+                        offchipLinkIds,
+                    });
+
+                    linkState.bpc = bpc;
+                    linkState.saturation = linkSaturation;
+                    epochState.linksPerNodeMap[nodeUid].saturation = nodeSaturation;
+                    epochState.linksPerNodeMap[nodeUid].offchipSaturation = offChipSaturation;
+                }
+            });
         });
     });
 };
 
-const recalculateLinkSaturation = (link: LinkState, totalOpCycles: number) => {
-    link.bpc = link.totalDataBytes / totalOpCycles;
-    link.saturation = (link.bpc / link.maxBandwidth) * 100;
+const recalculateLinkSaturation = ({
```

Comment:
the method is clearly doing more than recalculation of saturation, the name shoudl reflect this

Reply:
additionally we _may_ want to consider finally renaming saturation to congestion to bring ui and data model to the same terminology